thewalrus/operations.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Operations
+==========
+
+**Module name:** :mod:`thewalrus.operations`
+
+.. currentmodule:: thewalrus.operations
+
+Functions to construct the Fock representation of a Gaussian operation
+represented as a Symplectic matrix and complex displacements.
+
+
+Contains some Gaussian operations and auxiliary functions.
+
+Auxiliary functions
+-------------------
+
+.. autosummary::
+    n_two_mode_squeezed_vac
+    choi_expand
+
+Operations
+---------------
+
+.. autosummary::
+    fock_tensor
+"""
+
+from itertools import product
+import numpy as np
+from thewalrus import hafnian_batched
+from thewalrus.symplectic import expand
+from thewalrus.quantum import Amat
+
+
+def n_two_mode_squeezed_vac(n, r=np.arcsinh(1.0)):
+    r"""
+    Returns the symplectic matrix associated with two mode squeezing operations by amount r between modes i and i+n for 0<=i<n
```

Comment:
Better to move the details regarding `n` and `r` into the body of the docstring, or into the args list
```suggestion
    Returns the symplectic matrix associated with two mode squeezing operations.
```